src/main/java/fruitshop/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        Storage storage = new Storage();
+        int newQuantity = storage.getQuantity(fruitTransaction.getFruit())
+                - fruitTransaction.getQuantity();
+        if (newQuantity >= 0) {
+            storage.set(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded");
```

Comment:
The exception message should be more informative. Include information about the fruit and the attempted quantity that caused the negative balance.